rocketpy/stochastic/stochastic_model.py

```diff
@@ -0,0 +1,533 @@
+            tuples, lists, ints, floats, or None. The arguments will then be
+            validated and saved as attributes of the class in the correct
+            format. See each validation method for more information. None values
+            are allowed and will be replaced by the value of the attribute in
+            the main object. When saved as an attribute, the value will be saved
+            as a list with one item. If in the child class constructor an
+            argument of the original class is not allowed, then it has to be
+            passed as None in the super().__init__ call.
+
+        Raises
+        ------
+        AssertionError
+            If the input arguments do not conform to the specified formats.
+        """
+        self.object = object
+        self.last_rnd_dict = {}
+
+        for input_name, input_value in kwargs.items():
+            if input_name not in self.exception_list:
+                if input_value is not None:
+                    if "factor" in input_name:
+                        attr_value = self._validate_factors(input_name, input_value)
+                    elif input_name not in self.exception_list:
+                        if isinstance(input_value, tuple):
+                            attr_value = self._validate_tuple(input_name, input_value)
+                        elif isinstance(input_value, list):
+                            attr_value = self._validate_list(input_name, input_value)
+                        elif isinstance(input_value, (int, float)):
+                            attr_value = self._validate_scalar(input_name, input_value)
+                        else:
+                            raise AssertionError(
+                                f"'{input_name}' must be a tuple, list, int, or float"
+                            )
+                else:
+                    # if input_value is None, then the value will be taken from
+                    # the main object and saved as a one item list.
+                    attr_value = [getattr(self.object, input_name)]
+                setattr(self, input_name, attr_value)
+
+    def __str__(self):
+        s = ""
+        for key, value in self.__dict__.items():
+            if key.startswith("_"):
+                continue  # Skip attributes starting with underscore
+            if isinstance(value, tuple):
+                # Format the tuple as a string with the mean and standard deviation.
+                value_str = (
+                    f"{value[0]:.5f} ± {value[1]:.5f} "
+                    f"(numpy.random.{value[2].__name__})"
+                )
+            else:
+                # Otherwise, just use the default string representation of the value.
+                value_str = str(value)
+            s += f"{key}: {value_str}\n"
+        return s.strip()
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(object={self.object}, **kwargs)"
```

Comment:
if you want to "print(object)", the __str__ method will be used.
The __repr__ is (mainly) used for debugging purposes, it is usually a string that could be executed and yield the same value as the object.
The __str__ is a more "humanized" version, which is the case of your implementation of __str__ here.

I will comment the __repr__ method and leave a TODO to improve it in the future.
This way, the __repr__ will point to the __str__ method.

But just to raise this problem... When someone faces a bug, it will call the __str__ method, and this will be a hard to interpret sometimes.